### PR TITLE
Rewire dav files endpoint to home (better approach)

### PR DESCRIPTION
**Pull request by PVince81**
If the user specified in the dav files URL matches the current one,
rewire it to use the webDavHandler which is wired to the home storage.

This fixes path mapping issues.

Obsoletes https://github.com/cs3org/reva/pull/1120

- [x] EOS
   - [x] test regular file operations on dav files
   - [x] test upload on dav files
   - [x] test file ops on a received share
   - [x] test upload on a received share
- [x] BUG: litmus issue with OPTIONS on new dav
- [x] discard or update code from https://github.com/cs3org/reva/pull/1124 ? => discarded and rebased

### CODE REVIEW

**Comment by PVince81** [comment]
hmmm, so did my PR magically fix many issues ? @phil-davis 

image

**Comment by PVince81** [comment]
I've pushed a fix for the OPTIONS request, that should at least fix the Litmus tests

**Comment by PVince81** [comment]
there was a bug to the fallback, fixed now

expecting more tests to pass now

**Comment by PVince81** [comment]
local and litmus tests passed.

there are still a few trashbin tests that apparently were supposed to fail?
image

but I don't really see how this PR would help fixing those as it doesn't affect the trashbin endpoint

**Comment by PVince81** [comment]
Rebased onto master which has https://github.com/cs3org/reva/pull/1124 merged now.

I've also adjusted the expected failures. It is possible that the rewiring has fixed some issues very indirectly.

**Review by butonic — APPROVED** [review]

**Comment by butonic** [comment]
Cool!

File: tests/acceptance/expected-failures-on-OC-storage.txt [hunk 1]
```diff
@@ -500,13 +496,6 @@ apiTrashbin/trashbinFilesFolders.feature:241
 apiTrashbin/trashbinFilesFolders.feature:242
 apiTrashbin/trashbinFilesFolders.feature:255
 apiTrashbin/trashbinFilesFolders.feature:256
-apiTrashbin/trashbinFilesFolders.feature:273
-apiTrashbin/trashbinFilesFolders.feature:274
-apiTrashbin/trashbinFilesFolders.feature:275
-apiTrashbin/trashbinFilesFolders.feature:276
-apiTrashbin/trashbinFilesFolders.feature:277
-apiTrashbin/trashbinFilesFolders.feature:278
-apiTrashbin/trashbinFilesFolders.feature:289
```
Comment (phil-davis) [inline tests/acceptance/expected-failures-on-OC-storage.txt:509]: These are Scenario Outline Examples for the "new" dav-path. The Examples for the "old" dav-path were already passing. So the "rewiring" in this PR has fixed some issue with the "new" dav-path. That is "a good thing" (tm)

File: tests/acceptance/expected-failures-on-OC-storage.txt [hunk 2]
```diff
@@ -534,8 +523,6 @@ apiTrashbin/trashbinRestore.feature:206
 apiTrashbin/trashbinRestore.feature:207
 apiTrashbin/trashbinRestore.feature:279
 apiTrashbin/trashbinRestore.feature:280
-apiTrashbin/trashbinRestore.feature:295
-apiTrashbin/trashbinRestore.feature:310
```
Comment (phil-davis) [inline tests/acceptance/expected-failures-on-OC-storage.txt:538]: These are both Examples for "new" dav-path. The example for "old" dav-path at 294 and 309 were already passing.

**Review by phil-davis — APPROVED** [review]
All the trashbin scenarios that have started passing are for the "new" dav-path. So this PR definitely does fix stuff.

**Comment by phil-davis** [comment]
@labkode @butonic or ? IMO  this is "a good thing". Who can merge?

**Review by ishank011 — APPROVED** [review]

**Pull request merged**